src/Lykke.HttpClientGenerator/HttpClientGeneratorBuilderExtensions.cs

```diff
@@ -7,49 +7,31 @@ namespace Lykke.HttpClientGenerator
     /// </summary>
     public static class HttpClientGeneratorBuilderExtensions
     {
-        /// <summary>
-        /// Sets a name for this service which will be displayed in all errors
-        /// </summary>
-        ///
-        /// <param name="builder">The builder for the service</param>
-        /// <param name="serviceName">The name of the service</param>
-        ///
-        /// <returns>The service builder</returns>
-        ///
-        /// <exception cref="ArgumentNullException">
-        /// Thrown when <paramref name="builder"/> or <paramref name="serviceName"/> are null
-        /// </exception>
-        public static HttpClientGeneratorBuilder WithServiceName(
-            this HttpClientGeneratorBuilder builder,
-            string serviceName)
-        {
-            return AttachCallsWrapper<object>(builder, serviceName, showReason: false);
-        }
-
         /// <summary>
         /// Sets a name for this service which will be displayed in all errors,
-        /// additionally parsing and adding service errors
+        /// additionally parsing and adding service errors depending on parameter value
         /// </summary>
-        ///
+        /// 
         /// <param name="builder">The builder for the service</param>
         /// <param name="serviceName">The name of the service</param>
-        ///
+        /// <param name="showReason">Whether to add service errors or not</param>
         /// <typeparam name="T">
         /// The type representing the structure
         /// of the errors returned by the service
         /// </typeparam>
-        ///
+        /// 
         /// <returns>The service builder</returns>
-        ///
+        /// 
         /// <exception cref="ArgumentNullException">
         /// Thrown when <paramref name="builder"/> or <paramref name="serviceName"/> are null
         /// </exception>
         public static HttpClientGeneratorBuilder WithServiceName<T>(
             this HttpClientGeneratorBuilder builder,
-            string serviceName)
+            string serviceName,
+            bool showReason = true)
             where T : class
         {
-            return AttachCallsWrapper<T>(builder, serviceName, showReason: true);
+            return AttachCallsWrapper<T>(builder, serviceName, showReason);
         }
 
         private static HttpClientGeneratorBuilder AttachCallsWrapper<TApiError>(
```

Comment:
Let's make this public and rename to WithServiceName, so WithServiceName with 2 arguments will call the new one with 3 args